.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,160 @@
+          
+      - name: Create access token
+        id: token
+        env:
+          APP_ID: ${{ secrets.APP_ID }}
+          APP_PRIVATE_KEY: ${{ secrets.APP_PRIVATE_KEY }}
+        run: |
+          jwt=$(step crypto jwt sign --key /dev/fd/3 --issuer $APP_ID --expiration $(date -d +5min +%s) --subtle 3<<< $APP_PRIVATE_KEY)
+          installation_id=$(curl -s -H "Accept: application/vnd.github.v3+json" -H "Authorization: Bearer $jwt" https://api.github.com/app/installations | jq '.[] | select(.account.login == "${{ github.repository_owner }}") | .id')
+          token=$(curl -s -X POST -H "Accept: application/vnd.github.v3+json" -H "Authorization: Bearer $jwt" https://api.github.com/app/installations/$installation_id/access_tokens | jq -r '.token')
+          echo "::add-mask::$token"
+          echo "token=$token" >> $GITHUB_OUTPUT
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          token: ${{ steps.token.outputs.token }}
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.8"
+
+      - name: Install dependencies
+        run: pip3 install -r requirements.txt
+        
+      - uses: actions/download-artifact@v3
+        with:
+         path: artifacts
+
+      - name: Process data
+        shell: pwsh
+        run: |
+          Get-ChildItem -File -Path artifacts -Depth 1 | Foreach {. python3 process_results.py $_.fullname data.csv}
+
+      - name: Generate graph with Plotly for Python
+        if: ${{ false }}
+        shell: pwsh
+        run: |
+          . python3 generate_html.py data.csv index-python.html
+
+      - name: Get alerts
+        id: alerts
+        run: |
+          python3 generate_alert.py data.csv active_regressions.txt
+          if test -f "active_regressions.txt"; then
+            echo "file_exists=true" >> $GITHUB_OUTPUT
+            echo "file_content='$file_content'" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Create new issue if necessary
+        if: steps.alerts.outputs.file_exists == 'true'
+        run: |
+          numOpenIssues="$(gh api graphql -F owner=$OWNER -F name=$REPO -f query='
+            query($name: String!, $owner: String!) {
+              repository(owner: $owner, name: $name) {
+                issues(states:OPEN, filterBy: { labels:  ["active_regression"]}){
+                  totalCount
+                }
+              }
+            }
+          ' --jq '.data.repository.issues.totalCount')"
+
+          if [ $numOpenIssues -eq 0 ]; then
+            echo "Creating new issue"
+            gh issue create --title "Active NuGet restore regression" --label "active_regression" --body "\`\`\` $(cat active_regressions.txt)  \`\`\`" --repo $GITHUB_REPOSITORY
+          fi
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          OWNER: ${{ github.repository_owner }}
+          REPO: ${{ github.event.repository.name }}
+          
+      - uses: stefanzweifel/git-auto-commit-action@v4
```

Comment:
What exactly happens here? 
Is it like, after `data.csv` is generated, the action automatically commits it to the repository ? @marcin-krystianc

Reply:
Yes, the data.csv file contains benchmark results. It is being updated during this workflow and in this step it is auto-committed into the repository.